content/one-sided-concepts.rst

```diff
@@ -0,0 +1,206 @@
+
+
+At a glance: how does it work?
+------------------------------
+
+Let us look at the following figure, what routines are available in MPI for
+process 0 communicate a variable in its local memory to process 1?
+
+.. figure:: img/E02-steve-alice_step0.svg
+   :align: center
+
+   Steve, the sloth on the left, would like to send Alice, the sloth on the
+   right, the data in its ``Y`` variable. This data is stored in Steve's local
+   memory, depicted as a yellow box.
+
+It is foundational to MPI that every interaction between processes be
+*explicit*, so a simple assignment will not do.
+First, we must make a portion of memory on the *target process*, process 1
+in this case, visible for process 0 to manipulate.
+We call this a **window** and we will represent it as a blue diamond.
+
+.. figure:: img/E02-steve-alice_step1.svg
+   :align: center
+
+   We call collective routines, provided by MPI, to open a **memory window** on
+   each process in the communicator. Both the target and origin processes will
+   expose a portion of their memory through their respective windows.
+
+Once a *window* into the memory of process 1 is open, process 0 can access it and manipulate
+it. Process 0 can **put** (store) data in its local memory into the memory window of process
+1, using |term-MPI_Put|:
+
+.. figure:: img/E02-steve-alice_step2.svg
+   :align: center
+
+   The **origin process** (left sloth) puts data in the memory window of the
+   **target process** (right sloth).
+   The |term-MPI_Put| routine is represented with a red line whose arrowhead touches the
+   origin process of the call.
+
+In this example, process 0 is the origin process: it participates actively in
+the communication by calling the :term:`RMA` routine |term-MPI_Put|.  Process 1
+in the target process.
+
+Conversely, process 0 might have populated its memory window with some data: any
+other process in the communicator can now **get** (load) this data, using |term-MPI_Get|:
+
+.. figure:: img/E02-steve-alice_step3.svg
+   :align: center
+
+   The **origin process** (right sloth) gets data in the memory window of the
+   **target process** (left sloth).
+   The |term-MPI_Get| routine is represented with a blue line whose arrowhead touches the
+   origin process.
+
+In this scenario, process 1 is the origin process: it participates actively in the
+communication by calling the :term:`RMA` routine |term-MPI_Get|.  Process 0 is
+the target process.
+
+
+.. callout:: Graphical conventions
+
+   We have introduced these graphical conventions:
+
+   - A memory window is a blue diamond.
+   - A call to |term-MPI_Get| is a :blue:`blue` line whose arrowhead touches the origin
+     process.
+   - A call to |term-MPI_Put| is a :red:`red` line whose arrowhead touches the origin
+     process.
+   - For both routines, the direction of the arrowhead shows from which memory
+     window the data moves.
+
+
+.. challenge:: What kind of operations are being carried out?
```

Comment:
this multi-MCQ won't work well with Zoom since launching multiple polls after each other is clunky. But we can use hackMD for this so it's perfectly fine

Reply:
We can also move to breakout rooms.